### ISSUE TITLE
[java] False positives in UnusedNullCheckInEquals

### ISSUE THREAD

**Issue by Noah0120** [pmd/pmd#3432]
**Affects PMD Version:** 
6.36.0

**Rule:**
UnusedNullCheckInEquals
Please provide the rule name and a link to the rule documentation:
https://pmd.github.io/latest/pmd_rules_java_errorprone.html#unusednullcheckinequals

**Description:**
There are five cases that generate false positives:
1) Using `Arrays.equals()`
2) Using `Bytes.equals()`
    - The first and second cases don't use null checks.
   
3) Using `overridden equals() that has more than two arguments`
    - This case also doesn't need null checks for comparanda.
   
4) `.class.getName()` doesn't return null
    - According to `javadoc` of `class.getName()`, it doesn't return `null`.
       So, it is okay to use as the `caller` of `equals()`.
       Ref. https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html#getName--

5) null checked in `condition` of `conditional expression`
    - After null checked in `condition` part of `conditional expression`,
       PMD generates alarms.

**Code Sample demonstrating the issue:**

```
class UnusedNullCheckInEquals{
        void test(){
                //case1: Arrays.equals()
                if (other.name != null && java.util.Arrays.equals(other.name,name));
                //case2: Bytes.equals() 
                if (startRow != null && startRow.length > 0 && Bytes.equals(startRow,stopRow));

                //case3: oveerriden equals() that has more than two arguments
                if (k1 == k2 || (k1 != null && hashingStrategy.equals(k1, k2)));
                if (passthroughNsPrefix != null && 
                LexerUtils.equals(passthroughNsPrefix, attribute.namespacePrefix(), true, true));

                //case4: .class.getName() cannot return null
                if (driverDelegate != null && 
                StdJDBCDelegate.class.getName().equals(driverDelegate));
                
                //case5: null checked in condition of conditional expression
                if (this.start == other.start && this.length == other.length && 
                (this.file == null ? 
                 other.file == null : (other.file != null && this.file.equals(other.file))));    
        }
}
```

**Expected outcome:**
PMD shouldn't report anything. These are false positives.

Thank you for your consideration.

**Running PMD through:** *[CLI]*

**Comment by oowekyala** [comment]
- Case 1: this is not a false positive. You can remove the null check and your program won't change meaning. So the null check is indeed unused.
- Case 2: same
- Case 3: this is a false positive
- Case 4: see case 1. Whether Class.getName() returns a non null string is irrelevant
- Case 5: see case 1

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0. However, it will no longer flag null-checks prior to `Arrays.equals` or `Objects.equals`, so I'm opening a new issue for these FNs
